File: backend/semgrep_rules/tests/python/kisa_sw_13_http_response_splitting.py
def unsafe(request, response):
    value = request.GET.get("value")
    # ruleid: kisa.sw13.python.external-input-to-response-header
    response["X-User"] = value
    return response

def safe(response):
    # ok: kisa.sw13.python.external-input-to-response-header
    response["X-User"] = "fixed"
    return response

File: backend/semgrep_rules/tests/javascript/kisa_sw_13_http_response_splitting.js
function unsafe(req, res) {
  const value = req.query.value;
  // ruleid: kisa.sw13.javascript.external-input-to-response-header
  res.setHeader("X-User", value);
}
function safe(req, res) {
  // ok: kisa.sw13.javascript.external-input-to-response-header
  res.setHeader("X-User", "fixed");
}
